contribute/testing.md

```diff
@@ -0,0 +1,230 @@
+   </td>
+  </tr>
+  <tr>
+   <td>Triggers
+   </td>
+   <td>Correctness
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/util/TriggerTester.java">TriggerTester</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/AfterWatermarkTest.java">AfterWatermarkTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>Coders
+   </td>
+   <td>Encoding/decoding elements
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Serialization/deserialization of Coder
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Sizing of elements
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Deterministic
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Structural value equality
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+</table>
+
+
+## Beam Python Testing Matrix
```

Comment:
Hmm, realize we talk about Gearpump further down. Changed my mind -- testing is important and a criteria to graduate off a feature branch. So we should give us much detail as possible. ;-)